pkg/tecdsa/dkg/member.go

```diff
@@ -256,9 +259,382 @@ func (fm *finalizingMember) MarkInactiveMembers(
+// result hash.
+// The function assumes that the public key presented in the message is the
+// correct one. This key needs to be compared against the one used by network
+// client earlier, before this function is called.
+func (sm *signingMember) VerifyDKGResultSignatures(
+	messages []*dkgResultHashSignatureMessage,
+	signing chain.Signing,
+) (map[group.MemberIndex][]byte, error) {
+	// TODO: Consider moving the whole implementation to the resultSigner
+	//       interface
+
+	duplicatedMessagesFromSender := func(senderID group.MemberIndex) bool {
+		messageFromSenderAlreadySeen := false
+		for _, message := range messages {
+			if message.senderID == senderID {
+				if messageFromSenderAlreadySeen {
+					return true
+				}
+				messageFromSenderAlreadySeen = true
+			}
+		}
+		return false
+	}
+
+	receivedValidResultSignatures := make(map[group.MemberIndex][]byte)
+
+	for _, message := range messages {
+		// Check if message from self.
+		if message.senderID == sm.index {
+			continue
+		}
+
+		// Check if sender sent multiple messages.
+		if duplicatedMessagesFromSender(message.senderID) {
+			sm.logger.Infof(
+				"[member: %v] received multiple messages from sender: [%d]",
+				sm.index,
+				message.senderID,
+			)
+			continue
+		}
+
+		// Sender's preferred DKG result hash doesn't match current member's
+		// preferred DKG result hash.
+		if message.resultHash != sm.preferredDKGResultHash {
+			sm.logger.Infof(
+				"[member: %v] signature from sender [%d] supports result different than preferred",
+				sm.index,
+				message.senderID,
+			)
+			continue
+		}
+
+		// Check if the signature is valid.
+		ok, err := signing.VerifyWithPublicKey(
+			message.resultHash[:],
+			message.signature,
+			message.publicKey,
+		)
+		if err != nil {
+			sm.logger.Infof(
+				"[member: %v] verification of signature from sender [%d] failed: [%v]",
+				sm.index,
+				message.senderID,
+				err,
+			)
+			continue
+		}
+		if !ok {
+			sm.logger.Infof(
+				"[member: %v] sender [%d] provided invalid signature",
+				sm.index,
+				message.senderID,
+			)
+			continue
+		}
+
+		receivedValidResultSignatures[message.senderID] = message.signature
+	}
+
+	// Register member's self signature.
+	receivedValidResultSignatures[sm.index] = sm.selfDKGResultSignature
+
+	return receivedValidResultSignatures, nil
+}
+
+// shouldAcceptMessage indicates whether the given member should accept
+// a message from the given sender.
+func (sm *signingMember) shouldAcceptMessage(
+	senderID group.MemberIndex,
+	senderPublicKey []byte,
+) bool {
+	isMessageFromSelf := senderID == sm.index
+	isSenderValid := sm.membershipValidator.IsValidMembership(
+		senderID,
+		senderPublicKey,
+	)
+	isSenderAccepted := sm.group.IsOperating(senderID)
+
+	return !isMessageFromSelf && isSenderValid && isSenderAccepted
+}
+
+// SubmissionConfig contains parameters describing DKG submission process.
+type SubmissionConfig struct {
```

Comment:
I think that struct won't be needed eventually.

Reply:
Removed in b4b0f607f3967810ebd934fe99602b59ef8e7c80.